bindings/generator/generate.py

```diff
@@ -505,42 +505,44 @@ class ParsingPlaceholder:
     )
 
 
-def generate_client(api_specs: ApiSpecs) -> list[str]:
-    template = env.get_template("client_plugin_definitions.d.ts.j2")
-    output = (BASEDIR / "../../client/src/plugins/libparsec/definitions.d.ts").resolve()
-    print(f"Generating {output}")
-    # Don't use `write_text` given it outputs \r\n for newlines on Windows
-    output.write_bytes(template.render(api=api_specs).encode("utf8"))
-    return []
+def gen_generate_fn(
+    template: str, dest: Path, modified_crate: str | None = None
+) -> Callable[[ApiSpecs], str | None]:
+    dest = dest.resolve()
 
+    def _generate(api_specs: ApiSpecs) -> str | None:
+        _template = env.get_template(template)
+        print(f"Generating {dest}")
+        # Don't use `write_text` given it outputs \r\n for newlines on Windows
+        dest.write_bytes(_template.render(api=api_specs).encode("utf8"))
+        return modified_crate
 
-def generate_electron(api_specs: ApiSpecs) -> list[str]:
-    template = env.get_template("binding_electron_meths.rs.j2")
-    output = (BASEDIR / "../electron/src/meths.rs").resolve()
-    print(f"Generating {output}")
-    output.write_bytes(template.render(api=api_specs).encode("utf8"))
+    return _generate
 
-    return ["libparsec_bindings_electron"]
 
+generate_client = gen_generate_fn(
+    template="client_plugin_definitions.d.ts.j2",
+    dest=BASEDIR / "../../client/src/plugins/libparsec/definitions.d.ts",
+)
 
-def generate_electron_client(api_specs: ApiSpecs) -> list[str]:
-    template = env.get_template("binding_electron_index.d.ts.j2")
-    output = (BASEDIR / "../electron/src/index.d.ts").resolve()
-    print(f"Generating {output}")
-    output.write_bytes(template.render(api=api_specs).encode("utf8"))
-    return []
-
+generate_electron = gen_generate_fn(
+    template="binding_electron_meths.rs.j2",
+    dest=BASEDIR / "../electron/src/meths.rs",
+    modified_crate="libparsec_bindings_electron",
+)
 
-def generate_web(api_specs: ApiSpecs) -> list[str]:
-    template = env.get_template("binding_web_meths.rs.j2")
-    output = (BASEDIR / "../web/src/meths.rs").resolve()
-    print(f"Generating {output}")
-    output.write_bytes(template.render(api=api_specs).encode("utf8"))
+generate_electron_client = gen_generate_fn(
+    template="binding_electron_index.d.ts.j2", dest=BASEDIR / "../electron/src/index.d.ts"
+)
 
-    return ["libparsec_bindings_web"]
+generate_web = gen_generate_fn(
+    template="binding_web_meths.rs.j2",
+    dest=BASEDIR / "../web/src/meths.rs",
+    modified_crate="libparsec_bindings_web",
+)
 
 
-def generate(what: str, api_specs: ApiSpecs) -> list[str]:
+def generate(what: str, api_specs: ApiSpecs) -> str | None:
```

Comment:
In this function, we can replace the `generate_*` function calls with `generate_target` function. Eventually, the concrete arguments (`client_plugin_definitions.d.ts.j2`, and so on) can be defined somewhere in the file as constants.

Reply:
done

Reply:
The constant value is planned for a later PR (next to this one)